### PR TITLE
Fixes bug with adding hostname to web app.

### LINKED ISSUE #1984: 'az appservice web config hostname add' fails with NotFound
## Description

The command 'az appservice web config hostname add' fails with NotFound, and it appears as though an object representation of the web app is being passed to the service in lieu of the string name of the web app.

## Repro

```bash
$ az appservice web config hostname add -g DemoGroup --webapp AaronDemoHelloApp -n helloapp.twitchax.com
The Resource 'Microsoft.Web/sites/{'scm_site_also_stopped': False, 'site_name': u'AaronDemoHelloApp', 'resource_group': u'DemoGroup', 'client_cert_enabled': False, 'default_host_name': u'aarondemoh
elloapp.azurewebsites.net', 'host_names': [u'helloapp.twitchax.com', u'aarondemohelloapp.azurewebsites.net'], 'hosting_environment_profile': None, 'outbound_ip_addresses': u'40.78.17.157,40.78.21.1
91,40.78.22.38,40.78.21.180', 'id': u'' under resource group 'DemoGroup' was not found.
```

### PR DESCRIPTION
**Pull request by twitchax**
Fixes #1984.

### CODE REVIEW

**Comment by twitchax** [comment]
@johanste, do you think we can get this in soon? :)

My Azure CLI [blog post](http://blog.aaroney.com/2017/02/03/NET-Core-Web-App-in-Azure/) works better if this bug did not exist, haha.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -173,28 +173,27 @@ def show_container_settings(resource_group_name, name, slot=None):
 def _filter_for_container_settings(settings):
     return {x: settings[x] for x in settings if x in CONTAINER_APPSETTING_NAMES}
 
-def add_hostname(resource_group_name, webapp, name, slot=None):
+def add_hostname(resource_group_name, webappName, name, slot=None):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: This should be Python casing (i.e. webapp_name).

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -173,28 +173,27 @@ def show_container_settings(resource_group_name, name, slot=None):
...
     client = web_client_factory()
-    webapp = client.sites.get_site(resource_group_name, webapp)
-    binding = HostNameBinding(webapp.location, host_name_binding_name=name, site_name=webapp)
+    webapp = client.sites.get_site(resource_group_name, webappName)
+    binding = HostNameBinding(webapp.location, host_name_binding_name=name, site_name=webapp.name)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: So the problem was the parameter was redefined (webapp as a string in the method signature redefined to webapp the object). You could also use 'webapp_name' here too.

Comment (twitchax) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Good, good, will do.  I know the other methods were unaffected, but I wanted to keep them consistent.  I will update today.

**Comment by twitchax** [comment]
Thanks all!

**Pull request merged**